pkg/icingadb/overdue/sync.go

```diff
@@ -129,39 +132,12 @@ func (s Sync) log(ctx context.Context, objectType string, counter *com.Counter)
 	})
 }
 
-// luaGetOverdues takes the following KEYS:
-// * either icinga:nextupdate:host or icinga:nextupdate:service
-// * either icingadb:overdue:host or icingadb:overdue:service
-// * a random one
-//
-// It takes the following ARGV:
-// * the current date and time as *nix timestamp float in seconds
-//
-// It returns the following:
-// * overdue monitored objects not yet marked overdue
-// * not overdue monitored objects not yet unmarked overdue
-var luaGetOverdues = redis.NewScript(`
+//go:embed get_overdues.lua
+var getOverduesLua string
 
-local icingaNextupdate = KEYS[1]
-local icingadbOverdue = KEYS[2]
-local tempOverdue = KEYS[3]
-local now = ARGV[1]
-
-redis.call('DEL', tempOverdue)
-
-local zrbs = redis.call('ZRANGEBYSCORE', icingaNextupdate, '-inf', '(' .. now)
-for i = 1, #zrbs do
-	redis.call('SADD', tempOverdue, zrbs[i])
-end
-zrbs = nil
-
-local res = {redis.call('SDIFF', tempOverdue, icingadbOverdue), redis.call('SDIFF', icingadbOverdue, tempOverdue)}
-
-redis.call('DEL', tempOverdue)
-
-return res
-
-`)
+var luaGetOverdues = redis.NewScript(strings.TrimSpace(
+	regexp.MustCompile(`(?m)^--.*?$`).ReplaceAllString(getOverduesLua, ""),
```

Comment:
What's the `?` supposed to do in the regex? Also, if `EVALSHA` is used (from a quick look look at the lib I'm not entirely sure if this happens automatically or only if you call `s.Load()` on the script first), it wouldn't really matter if a few extra comments are sent over the wire once.

Reply:
1. Explicitly includes only up to the end of the current line.
2. No such automation here.
3. Less (data) is more.

Reply:
1. TIL: `*` longest possible match, `*?` shortest possible match.
2. (intentionally left blank, just there to make the list numbering work out)
3. Well, then `EVALSHA` might be a good idea, but time for another PR.

Reply:
1. -
2. -
3. No, I meant no automation like my regex.

Reply:
3. From a quick look at the code, `script.Run()` attempts to do `EVALSHA` but if that fails, it falls back to `EVAL` instead of loading the script, so seems like we're sending the script every time at the moment (and waste one round trip trying `EVALSHA`)

Reply:
> we're sending the script every time at the moment

Huh?

Reply:
Ah, I wasn't aware that `EVAL` implicitly caches the script, I thought you'd need an explicit `SCRIPT LOAD` for that. So never mind what I've said (would probably be too easy if https://redis.io/commands/eval/ mentioned any of the words "cache", "hash", "sha1").